homeassistant/components/climate/warmup4ie.py

```diff
@@ -0,0 +1,202 @@
+"""platform that offers a connection to a warmup4ie device."""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import PLATFORM_SCHEMA, ClimateDevice
+from homeassistant.const import (
+    ATTR_TEMPERATURE, CONF_NAME, CONF_PASSWORD, CONF_USERNAME, CONF_ROOM,
+    TEMP_CELSIUS)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.climate.const import (SUPPORT_TARGET_TEMPERATURE,
+                                                    SUPPORT_AWAY_MODE,
+                                                    SUPPORT_OPERATION_MODE,
+                                                    SUPPORT_ON_OFF, STATE_AUTO,
+                                                    STATE_MANUAL)
+
+
+REQUIREMENTS = ['warmup4ie==0.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_AWAY_MODE |
+                 SUPPORT_OPERATION_MODE | SUPPORT_ON_OFF)
+
+CONF_LOCATION = 'location'
+CONF_TARGET_TEMP = 'target_temp'
+
+
+DEFAULT_NAME = 'warmup4ie'
+DEFAULT_TARGET_TEMP = 20
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_LOCATION): cv.string,
+    vol.Required(CONF_ROOM): cv.string,
+    vol.Optional(CONF_TARGET_TEMP,
+                 default=DEFAULT_TARGET_TEMP): vol.Coerce(float),
+})
+
+
+# pylint: disable=no-member
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Demo climate devices."""
+    name = config.get(CONF_NAME)
+    user = config.get(CONF_USERNAME)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.